doc/developer/design/20230814_unbilled_replicas.md

```diff
@@ -0,0 +1,203 @@
+* Introduce replica sets.
+* Add a billing property to replica sets.
+* Record replica set properties in audit events.
+
+## Non-Goals
+
+We do not plan to do the following:
+* Allow customers general access to replica sets.
+
+## Overview
+
+We introduce the concept of replica sets, which capture partitions
+of replicas sharing configuration aspects. Replica sets can co-exist, and a default
+replica set captures the current configuration of a cluster. Each replica is a member of a single
+replica set.
+
+At this time, we only support replica sets for internal users, but do not
+give customers the permission to directly interact with replica sets. This
+is to give us time to validate the feature, and decide on reserved names.
+
+## Detailed description
+
+We extend clusters to support multiple replica configurations, and name each a replica set.
+A replica set describes the properties of a partition of replicas within a cluster.
+A cluster can have several replica sets, and each replica is part of a single replica set.
+
+Multiple replica set within a cluster can co-exist simultaneously, and can have different parameters.
+We do not change that all replicas in a cluster execute equivalent commands and participate equally in producing responses.
+
+```sql
+-- Creating a managed replica set without billing
+CREATE REPLICA SET cluster.support WITH (SIZE '3xlarge', BILLED AS 'unbilled');
+-- Drop the replica set, releasing any associated resource.
+DROP REPLICA SET cluster.support;
+
+-- Alternatively, create an unmanaged replica set without billing.
+CREATE REPLICA SET cluster.support WITH (MANAGED = FALSE, BILLED AS 'unbilled');
+-- Create a replica in the replica set.
+CREATE CLUSTER REPLICA cluster.r1 SIZE '3xlarge' IN REPLICA SET support;
+-- Drop the replica set, also dropping replica `r1`.
+DROP REPLICA SET cluster.support CASCADE;
+```
+
+### Replica sets and replicas
+
+At the moment, clusters only contain replicas.
+With the introduction of replica sets, we change this property to allow different kinds of objects as part of a cluster.
+Replicas and replica sets share a common cluster namespace, so their names need to be unique within a cluster, but can be reused across clusters.
+
+At this time, users do not need to work with replica sets if they don't choose to.
+Each cluster has an implicit replica set (a.k.a, the `NULL` replica set?) that they interact with using the managed or unmanaged cluster SQL API.
+
+The following example demonstrates using the implicit replica set:
+```sql
+CREATE CLUSTER clname SIZE '3xlarge';
+```
+
+### Resources within replica sets
+
+Replicas are part of a replica sets, and their lifetime is tied to that of the replica set:
+
+```sql
+-- Cluster `cl` has an implicit replica set with one replica `r1`
+DROP REPLICA SET cl.default;
+ERROR: Replica set cl.default has objects
+HINT: Drop objects first, or specify CASCADE.
+
+DROP REPLICA SET cl.default CASCADE;
+```
```

Comment:
What happens after you drop the default replica set? Chaos? Inability to say `ALTER CLUSTER cluster (SIZE = 'small')` until you recreate the default replica set?